src/cudamatrix/cu-sparse-matrix.h

```diff
@@ -43,82 +44,63 @@ Real TraceMatSmat(const CuMatrixBase<Real> &A,
                   const CuSparseMatrix<Real> &B,
                   MatrixTransposeType trans = kNoTrans);
 
-template <class Real>
+template<class Real>
 class CuSparseMatrix {
- public:
-  friend class CuMatrixBase<float>;
-  friend class CuMatrixBase<double>;
-  friend class CuMatrixBase<Real>;
-  friend class CuVectorBase<float>;
-  friend class CuVectorBase<double>;
-  friend class CuVectorBase<Real>;
+public:
+  friend class CuMatrixBase<float> ;
+  friend class CuMatrixBase<double> ;
+  friend class CuMatrixBase<Real> ;
+  friend class CuVectorBase<float> ;
+  friend class CuVectorBase<double> ;
+  friend class CuVectorBase<Real> ;
 
   friend Real TraceMatSmat<Real>(const CuMatrixBase<Real> &A,
                                  const CuSparseMatrix<Real> &B,
                                  MatrixTransposeType trans);
 
-  MatrixIndexT NumRows() const { return num_rows_; }
+  MatrixIndexT NumRows() const;
 
-  MatrixIndexT NumCols() const { return num_cols_; }
+  MatrixIndexT NumCols() const;
 
   MatrixIndexT NumElements() const;
 
-  template <typename OtherReal>
-  void CopyToMat(CuMatrixBase<OtherReal> *dest,
-                 MatrixTransposeType trans = kNoTrans) const;
+  template<typename OtherReal>
+  void CopyToMat(CuMatrixBase<OtherReal> *dest, MatrixTransposeType trans =
+                     kNoTrans) const;
 
   Real Sum() const;
 
   Real FrobeniusNorm() const;
 
-  // returns pointer to element data, or NULL if empty (use with NumElements()).
-  // This should only be called when CUDA is enabled.
-  MatrixElement<Real> *Data();
-
-  // returns pointer to element data, or NULL if empty (use with NumElements()),
-  // const version. This should only be called when CUDA is enabled.
-  const MatrixElement<Real> *Data() const;
-
   /// Copy from CPU-based matrix.
-  CuSparseMatrix<Real> &operator = (const SparseMatrix<Real> &smat);
+  CuSparseMatrix<Real> &operator =(const SparseMatrix<Real> &smat);
 
   /// Copy from possibly-GPU-based matrix.
-  CuSparseMatrix<Real> &operator = (const CuSparseMatrix<Real> &smat);
+  CuSparseMatrix<Real> &operator =(const CuSparseMatrix<Real> &smat);
+
+  void Resize(const MatrixIndexT num_rows, const MatrixIndexT num_cols,
```

Comment:
Defaulting to 0 was trying to be compatible with other classes, by using two arguments only, but probably not reasonable. I will add a comment for it.